### ISSUE TITLE
How to collaborate?

### ISSUE THREAD

**Issue by jessecusack** [mousebrains/RocklandAPI#3]
@mousebrains, how do you want to work on this. Would you prefer I make pull-requests from my fork or that I commit directly to this repo or something else?

**Comment by mousebrains** [comment]
I'm currently looking at refactoring how projects and profiles are stored internally. Projects contains a dictionary of Project objects which contains Profiles which is a list/dict of Profile objects. 

The items that need updated include:

- Save the project token that is returned when a project is created. 
- Save the profile token that is returned when a profile is uploaded.
- Implement Project/Profile which takes a ProjectToken and returns a list of profile tokens. 
- Get data give a project and profile tokens.

**Comment by mousebrains** [comment]
I've toying with saving the objects in an SQLite database, but I'm concerned with concurrent writes.

**Comment by mousebrains** [comment]
I'm done typing for a while, so have at it saving thinks to the main branch.

**Comment by jessecusack** [comment]
Ok, thank you for the summary. I'm still getting up to speed with the code and won't be able to spend much time on it until I'm less busy in a day or two. 

I'm not familiar with SQL so would be slow to figure that out.

Somewhat related, is there a plan for organizing the q files locally? and then the processed version upon download?

**Comment by mousebrains** [comment]
The q-files will be pulled over Iridium by SFMC, which places the files in a deployment/glider directory, along with all the other files sent to/pulled from the glider on gliderfmc0. The gliderfmc0 files will be stored on gliderfs2 in a raw files directory forever. I have not gotten to the q-files/results files yet, but I assume it will be on gliderfs2 in a cooked directory tree for each deployment/glider.

**Comment by jessecusack** [comment]
I assume gliderfs2 is a NAS in CEOAS?

**Comment by mousebrains** [comment]
No, gliderfs2 is a VM on glidervms0 in the ECC data center in Burt Hall. Gliderfs2 is running RAID 1. Both gliderfs2 and gliderfmc0 are backed up daily to two NASes running RAID6 in the ECC data center. There is also a backup to gliderdata VM on gliderfmc1 with a RAID1 array with a ZFS file system with daily snapshots. I don't believe in loosing data.

We use gliderfs2 for various processing too.
